domain/src/main/java/me/androidbox/domain/authentication/usecase/LoginUseCase.kt

```diff
@@ -0,0 +1,8 @@
+package me.androidbox.domain.authentication.usecase
+
+import me.androidbox.domain.authentication.NetworkResponseState
+import me.androidbox.domain.authentication.model.LoginModel
+
+fun interface LoginUseCase {
+    suspend fun execute(email: String, password: String): NetworkResponseState<LoginModel>
+}
```

Comment:
I'm still not fully understanding why the usecase is an interface
Use cases can be a simply class and that's it, since you won't be implementing this interface on testing or anywhere else

If an interface is going to be implemented only once, then there's no need to have an interface

For example, Repository, you'll implement it twice. One on the production code, and one for testing
For use cases, this isn't going to happen

Reply:
@MKiperszmid @philipplackner 
This has been a familiar pattern that I have followed. Basically, I prefer to program to an interface. One reason is that I prefer to inject the interface into a constructor and use @Binds to bind it to an implementation.  Its easier to mock interfaces than concrete implementations. But I guess this can be open to discussion. But I value your reviews to have a second opinion on this.